### ISSUE TITLE
Comment bubbles positioning is inconvenient

### ISSUE THREAD

**Issue by BeksOmega** [mit-cml/appinventor-sources#2180]
**Describe the bug**

How comment bubble positioning works is not very convenient.

Firstly, when you open a comment it will often position itself offscreen where you can't see it. This is obviously not helpful haha. I have previously fixed this behavior in blockly core, and I would be happy to move those changes over.

Secondly, this bad positioning means you often want to move your comment to a better position. But if you close and reopen the comment, it does not stay where you moved it, which is annoying.

![CommentPositioning](https://user-images.githubusercontent.com/25440652/80547202-7fb0fc00-896c-11ea-9af9-9fb70c00cad8.gif)

**Affects**

- [X] Blocks editor

**Expected behavior**

Option 1:
I expect the comment bubble to stay where I positioned it previously.

If the block is moved while the comment is closed, when I reopen the comment it should be in the same position *relative to the block* (not the workspace). This will make it match the "tracking" behavior a comment has while it is open:
![CommentTracking](https://user-images.githubusercontent.com/25440652/80547195-7b84de80-896c-11ea-91ce-2e8dc54b3e7a.gif)

Note: The  behavior detailed in this option does *not* currently exist in core.

Option 2:
If we add the behavior detailed in option 1, any bubbles that have been moved by the user will lose the ability to avoid covering up the block they are attached to. This avoiding behavior is demonstrated in [this video](https://www.youtube.com/watch?v=JkVLbHiKtFk&feature=youtu.be&t=20). And I think it's pretty convenient and nice to have.

It may be better to just make bubble positioning better (such as how it currently works is in blockly core -- see [video](https://www.youtube.com/watch?v=JkVLbHiKtFk&feature=youtu.be)) so that people aren't moving around their bubbles as much. That way we don't need to implement option 1 and we can keep the avoiding behavior.

**Steps to reproduce**

1) Make a block that's really tall and wide (like in the gif above)
1) Add a comment to a block.
2) Observe how the comment goes off screen.
2) Move the comment bubble.
3) Close the comment.
4) Reopen the comment.
5) Observe how the comment is not in the same place you left it.

**Additional Info**

This problem was originally discussed in [this forum post](https://community.appinventor.mit.edu/t/comment-font-size-is-so-small/7847).

I think both options detailed above have their pros and cons, personally I don't have a strong preference. I just want to make the blocks editor more convenient for end-users so I'm happy to build either option =)

It may be worth it to build both options, and then test them with end users, to see which option is better.

**Comment by ewpatton** [comment]
@BeksOmega I think for now we should just try to bring it in line with what core does.

**Comment by BeksOmega** [comment]
@ewpatton Sounds good! Should I get started on that, or should I wait until after the meeting on the 7th? (I saw there was going to be discussion of a Blockly upgrade)

**Comment by ewpatton** [comment]
@BeksOmega Let's wait on that. @ellelili2025 has been working on the upgrade, so once that work is done we should gain your functionality from core.

**Comment by BeksOmega** [comment]
> @ellelili2025 has been working on the upgrade, so once that work is done we should gain your functionality from core.

Oh sweet that's awesome! Have you thought about moving to the node distribution? It would be fun to move some things like the typeblock system over to plugins :D

**Comment by ewpatton** [comment]
@BeksOmega I don't think we will be able to move to using Blockly in any other way until core has enough extension points that we no longer need to monkey patch or rewrite parts of core to make it work with App Inventor.

**Comment by BeksOmega** [comment]
> I don't think we will be able to move to using Blockly in any other way until core has enough extension points that we no longer need to monkey patch or rewrite parts of core to make it work with App Inventor.

@ewpatton Darn, well that's pretty unlikely to happen :/ I get where you're coming from though. It's definitely easier to "manually" upgrade blockly periodically than to start over from scratch.

But it got me thinking - the Blockly team /has/ to have a recommendation for repos in this situtation. I mean it was so common to fork Blockly before, that I'm pretty sure Sam would have some thoughts on this.

So I went ahead and created a [post](https://groups.google.com/forum/#!topic/blockly/EI4f9agMufA) to ask about it. As I said over there, I know App Inventor would have other problems with moving over besides just the monkey-patching, but I'm still curious what they have to say about it.

**Comment by ewpatton** [comment]
This no longer seems to be the case since we upgraded to Blockly 10.
